internal/service/ssoadmin/customer_managed_policy_attachment.go

```diff
@@ -0,0 +1,247 @@
+package ssoadmin
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssoadmin"
+	"github.com/hashicorp/aws-sdk-go-base/v2/awsv1shim/v2/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+const (
+	customerPolicyAttachmentTimeout = 5 * time.Minute
+)
+
+func ResourceCustomerManagedPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCustomerManagedPolicyAttachmentCreate,
+		Read:   resourceCustomerManagedPolicyAttachmentRead,
+		Delete: resourceCustomerManagedPolicyAttachmentDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+
+			"instance_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"permission_set_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"customer_managed_policy_reference": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+						},
+						"path": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "/",
+							ForceNew: true,
```

Comment:
As per https://docs.aws.amazon.com/singlesignon/latest/APIReference/API_CustomerManagedPolicyReference.html
```suggestion
							ForceNew: true,
							ValidateFunc: validation.StringLenBetween(0, 512),
```